crates/sqlite_store/src/store.rs

```diff
@@ -0,0 +1,1095 @@
+use rusqlite::{named_params, Connection, Error};
+use serde::{Deserialize, Serialize};
+use std::collections::{BTreeMap, BTreeSet};
+use std::default::Default;
+use std::marker::PhantomData;
+use std::str::FromStr;
+use std::{fmt::Debug, path::Path};
+
+use bdk_chain::bitcoin::consensus::{deserialize, serialize};
+use bdk_chain::bitcoin::hashes::Hash;
+use bdk_chain::bitcoin::{BlockHash, Network, OutPoint, ScriptBuf, Transaction, TxOut, Txid};
+use bdk_chain::miniscript::{Descriptor, DescriptorPublicKey};
+use bdk_chain::{
+    indexed_tx_graph, keychain, local_chain, tx_graph, Anchor, Append, BlockId,
+    ConfirmationHeightAnchor, ConfirmationTimeHeightAnchor, DescriptorExt, DescriptorId,
+    PersistBackend,
+};
+
+use crate::{ChangeSet, ReadError, WriteError};
+
+const MIGRATIONS: &[&str] = &[
+    // schema version control
+    "CREATE TABLE version (version INTEGER) STRICT;",
+    "INSERT INTO version VALUES (1);",
+    // network is the valid network for all other table data
+    "CREATE TABLE network
+    (name TEXT UNIQUE NOT NULL)
+    STRICT;",
+    // keychain is the json serialized keychain structure as JSONB,
+    // descriptor_id is a sha256::Hash id of the descriptor string w/o the checksum,
+    // descriptor is the complete descriptor string,
+    // last active index is a u32
+    "CREATE TABLE keychain
+    (keychain BLOB PRIMARY KEY NOT NULL,
+     descriptor TEXT UNIQUE NOT NULL,
+     descriptor_id BLOB UNIQUE NOT NULL,
+     last_revealed INTEGER)
+    STRICT;",
+    // hash is block hash hex string,
+    // block height is a u32,
+    // previous block hash hex string
+    "CREATE TABLE block
+    (hash TEXT PRIMARY KEY NOT NULL,
+     height INTEGER NOT NULL)
+    STRICT;",
+    // txid is transaction hash hex string (reversed)
+    // whole_tx is a consensus encoded transaction,
+    // last seen is a u64 unix epoch seconds
+    "CREATE TABLE tx
+    (txid TEXT PRIMARY KEY NOT NULL,
+     whole_tx BLOB NOT NULL,
+     last_seen INTEGER)
+    STRICT;",
+    // Outpoint txid hash hex string (reversed)
+    // Outpoint vout
+    // TxOut value as SATs
+    // TxOut script consensus encoded
+    "CREATE TABLE txout
+    (txid TEXT NOT NULL,
+     vout INTEGER NOT NULL,
+     value INTEGER NOT NULL,
+     script BLOB NOT NULL,
+     PRIMARY KEY (txid, vout))
+    STRICT;",
+    // set of block "anchors" and txids
+    // block hash hex string
+    // block height u32
+    // confirmation height u32
+    // confirmation time unix epoch seconds
+    // txid is transaction hash hex string (reversed)
+    "CREATE TABLE anchor
+    (block_hash TEXT NOT NULL REFERENCES block (hash),
+     confirmation_height INTEGER,
+     confirmation_time INTEGER,
+     txid TEXT NOT NULL REFERENCES tx(txid))
+    STRICT;",
+];
```

Comment:
What if we move this to `.sql` file?
We get syntax highlighting and maybe some IDE introspection and autocomplete?

Reply:
I'd love to get this out of the code but to include it in a .sql file I'd have to bring in another dependency that includes static files into the code and I'd rather not do that just for better IDE support.

Reply:
What about?
```rust
const MIGRATIONS: &str = include_str!("path/to/schema.sql");
```

Reply:
thanks will try this approach!